tasks/main.yml

```diff
@@ -32,6 +32,26 @@
   changed_when: False
   failed_when: False
 
+- name: singularity | remove older Go installation
+  file:
+    path: '/opt/go/{{ item }}'
+    state: absent
+  with_items: "{{ old_go_versions }}"
```

Comment:
You can do better than listing all the possible old version 😛 
Maybe by:
 - listing and put the result in a variable using `register`
 - then filter to exclude the current one with https://jinja.palletsprojects.com/en/2.10.x/templates/#filters

Reply:
With [find_module](https://docs.ansible.com/ansible/latest/modules/find_module.html), you can list all the GO versions installed on the filesystem.

Reply:
I didn't need to filter, I used regex to exclude the current go installation